### ISSUE TITLE
If this were a real 80s CPU ..

### ISSUE THREAD

**Issue by skx** [skx/go.vm#1]
We have a lot of special purpose instructions/opcodes which are fun to play with but which wouldn't exist in a real processor, for example:

* `MEMCPY`
   * Could be implemented via PEEK/POKE and a loop
* `SYSTEM`
   * Can't be implemented by hand, but isn't so terribly useful
   * At least in part because the results are output to the console and not stored in a register.

We're also missing the ability to carry out some operations, either in a simple fashion, or at all:

* We can't convert a string to uppercase easily
  * Though we could iterate over some memory and cmp >= 'a' && <= 'z'
  * We can't iterate over the characters in a string stored in a register though, or modify them in-place.
* We can't input string from the user into a register/RAM.
* We can't draw to the console.
   * Meaning we can't write "tetris" or other simple games.

If this were a real CPU we'd call into fixed addresses into ROM to get support for things, or we'd have an interrupt function.  For example we could define the `INT 0xNN` instruction.  Then using that we could provide facilities

* INT 0x00
   * Read a string from the user, store in register #0.
* INT 0x01
   * Output a single character, as stored in #0
* INT 0x02
   * Return the length of the string stored in #0
* INT 0x03
   * Draw the character stored in #0 at coordinates #1,#2
* ..

In short we could add higher-level useful utilities in a way that didn't require the addition of more single-purpose instructions.  I don't have a strong view of what things are missing, because I've never tried to write "complex" programs for this CPU, but if we're all about the learning then it should be possible to do so!

**Comment by tay10r** [comment]
The `MEMCPY` instruction will probably perform better using the native CPU instead of `PEEK` and `POKE`.

If you were actually designing hardware, it'd be better implemented by those two instructions. Since it's explicitly a virtual machine, I'd say leave it as is.
